src/string-ext.ts

```diff
@@ -266,4 +336,14 @@ Object.defineProperty(String.prototype, "hexDecode", {
     {
         return StringExt.hexDecode(this.toString());
     }
+});
+
+Object.defineProperty(String.prototype, "matchesFormat", {
+    configurable: false,
+    enumerable: false,
+    writable: false,
+    value: function (format: string): boolean
+    {
```

Comment:
if(format == null || typeof format !== "string" ||  format.trim() === "") throw new Error("format must be a valid string");

Reply:
format also cannot contain more than one wildcard char in it. validate this (dont forget to account for escape sequence).

Reply:
I check multiple `*` later on when I create tokens